types/block.go

```diff
@@ -32,7 +32,7 @@ type (
 		Transactions []Transaction   `json:"transactions"`
 	}
 
-	// A BlockHeader, when encoded, is an 80-byte constant size field
+	// BlockHeader when encoded, is an 80-byte constant size field
```

Comment:
Interesting. `golint` doesn't care about the `A`, but it does care about the `,`. I would rephrase this as:
```
A BlockHeader contains the data that, when hashed, produces the Block's ID.
```